libshaderc/src/shaderc.cc

```diff
@@ -449,14 +449,17 @@ void shaderc_compile_options_set_hlsl_register_set_and_binding(
 }
 
 shaderc_compiler_t shaderc_compiler_initialize() {
-  static shaderc_util::GlslangInitializer* initializer =
+  shaderc_util::GlslangInitializer* initializer =
```

Comment:
This will unfortunately fail in the case where you have 2 instances of the compiler at once.
I absolutely understand and agree with what you want to do, and why.

I think the right solution here would be to have the GlslangInitializer be basically a static refcount. In shaderc_compiler_release we decrement the refcount, and clear the state.

There would need to be a bit of extra locking but I thin it should work in the general case.